### PR TITLE
[enhancement] Implemented register_menu_subitem #230

### LINKED ISSUE #230: [enhancement] Register sub group item
**Discussion**
https://github.com/openwisp/openwisp-controller/pull/513#pullrequestreview-722471186

### PR DESCRIPTION
**Pull request by niteshsinha17**
### Changes

Now we can register an item in a registered group.
Tests are also included.

### Checklist

- [x] I have read the [contributing guidelines](http://openwisp.io/docs/developer/contributing.html#how-to-commit-your-changes-properly).
- [x] I have manually tested the proposed changes.
- [x] I have written new test cases to avoid regressions. (if necessary)
- [x] I have updated the documentation. (e.g. README.rst)
- [x] I have added [change!] to commit title to indicate a backward incompatible change. (if required)
- [x] I have checked the links added / modified in the documentation.

closes #230

### CODE REVIEW

File: openwisp_utils/admin_theme/menu.py [hunk 1]
```diff
@@ -211,6 +211,50 @@ def register_menu_group(position, config):
...
+    else:
+        # Unknown
+        raise ImproperlyConfigured(
+            f'Invalid config "{config}" provided for sub group item'
+        )
+    if item_position in group.items:
+        name = group.items[item_position]
+        raise ImproperlyConfigured(
+            f'A group item with config {config} is being registered at position\
+            "{item_position}" in a group but another item named "{name}" is already present their.'
```
Comment (nemesisdesign) [inline openwisp_utils/admin_theme/menu.py:253]: is already present their. > is already registered at the same position.

**Review by nemesisdesign — CHANGES_REQUESTED** [review]
Only one minor change requested below.

@pniaps take a look if you can.

File: openwisp_utils/admin_theme/menu.py (hunk 1 above)
Comment (niteshsinha17) [reply openwisp_utils/admin_theme/menu.py]: Done

**Review by nemesisdesign — CHANGES_REQUESTED** [review]
Looks good! 
Can you please make sure these 3 lines are covered in the test so we avoid the coveralls failure?
https://coveralls.io/jobs/85396242/source_files/11592866249#L241

**Comment by niteshsinha17** [comment]
> Looks good!
> Can you please make sure these 3 lines are covered in the test so we avoid the coveralls failure?
> https://coveralls.io/jobs/85396242/source_files/11592866249#L241

@nemesisdesign Thanks. I have added them

**Review by nemesisdesign — APPROVED** [review]

**Pull request merged**